File: Software Development Module/Revision/Airline/AirlineApp.java
/*
AirlineApp.java
Billy Cussen
29/07/2020
*/

import java.util.Scanner;
import java.util.InputMismatchException;

public class AirlineApp{

    public static void main(String[]args){

        //Variables
        String status = new String();
        int flights = 0;
        String pointsEarned;
        boolean inputCheckStatus;
        boolean inputCheckFlights;

        //Object
        Scanner keyboard;
        Airline myAirline;

        //Input
        do{
            keyboard = new Scanner(System.in);
            inputCheckStatus = true;

            try{
                System.out.print("Please enter your Loyalty Status(Silver, Gold or Platinum): ");
                status = keyboard.nextLine();

                if(!(status.equalsIgnoreCase("Silver")||status.equalsIgnoreCase("Gold")||status.equalsIgnoreCase("Platinum"))){
                    throw new InputMismatchException();
                }

            }catch(InputMismatchException e){
                System.out.println("You need to enter Silver, Gold or Platinum!");
                inputCheckStatus = false;
            }finally{
                keyboard = null;
            }

        }while(!inputCheckStatus);

        do{
            keyboard = new Scanner(System.in);
            inputCheckFlights = true;

            try{
                System.out.print("Please enter your Total flights with us this Calender Year: ");
                flights = keyboard.nextInt();

                if(flights<0){
                    throw new InputMismatchException();
                }

            }catch(NumberFormatException | InputMismatchException e){
                System.out.println("You need to enter a number (e.g. 1, 2 3)!");
                inputCheckFlights = false;
            }finally{
                keyboard = null;
            }

        }while(!inputCheckFlights);

        //Compute
        myAirline = new Airline(status, flights);
        pointsEarned = myAirline.getPointsEarned();
        myAirline = null;

        //Output
        System.out.println(pointsEarned);
    }

}

File: Python/Software Development Module - Python/Revision/Airline/AirlineApp.py
"""
AirlineApp.py
Billy Cussen
20/10/2020
"""
#Objects
import Airline
from Airline import myAirline
import gc

#Variables
repeat = "y"

while repeat == "y":

    inputCheck = False

    #Input - Membership?
    while inputCheck == False:
        try:
            membership = str(input("Please enter your Membership Level (\"Gold\", \"Silver\" or \"Bronze\"): "))
            membership = membership.lower()
            if membership != "gold" and membership != "silver" and membership != "bronze":
                raise Exception
        except Exception as e:
            print("You need to enter Gold, Silver or Bronze")
        else:
            inputCheck = True

    #Input - Flights this year
    while inputCheck:

        try:
            flights = int(input("Please enter your your total flights with us this year: "))
            if flights < 0:
                raise Exception
        except Exception as e:
            print("You need to enter a Valid Number (e.g. 0, 1, 2)!")
        else:
            inputCheck = False

    #Process and Output
    myAirlineApp = myAirline(membership,flights)
    print(myAirlineApp.getPoints())
    myAirlineApp = None
    gc.collect()

    repeat = ""
    while repeat.lower() != "y" and repeat.lower() != "n":
        repeat = str(input("Please enter \"y\" to repeat and \"n\" to exit: "))
        if repeat.lower() != "y" and repeat.lower != "n":
            print("You need to enter \"y\" or \"n\"")

print("Thanks for using my Airline App")